### PR TITLE
Support to deploy backend services to Azure serverless

### PR DESCRIPTION
**Pull request by sjuarezgx**
Add a new deploy target to be able to support the deployment of the apps's backend services to Azure as http functions.
Issue:90458

### CODE REVIEW

File: src/AzureServerless/Properties.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	    </Childs>
+    </Group>
+    <Group>
+      <Id>AzServerlessSettings</Id>
+      <Name>Serverless Deploy Settings</Name>
+      <Description>Serverless deploy settings</Description>
+      <Type>Main</Type>          
+        <Childs>
+          <Prop>
+            <Id>AZURE_SERVERLESS_OPENAPI_FILE</Id>
```
Comment (ggallotti) [inline src/AzureServerless/Properties.xml:72]: What do we expect in this Property? We need to specify an open api file? Which one?

I thought that we would generate it automatically.

File: src/AzureServerless/deploy.msbuild [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	<PropertyGroup>
+		<GXDeployFileProject>$([System.IO.Path]::GetFullPath('$(DeployFullPath)\..\..\..\..\web\$(ProjectName).gxdproj'))</GXDeployFileProject>
+	</PropertyGroup>
+	<Import Project="$(GX_PROGRAM_DIR)\GeneXus.AzureDeploy.targets"/>
+	<Import Project="$(GXDeployFileProject)"/>
+		
+	<PropertyGroup Condition="'$(GENERATOR)' == '.NET Core'">
+		<UploadSource>$(DeployFullPath)</UploadSource>
+		<DeploymentTargetBinPath>$(GX_PROGRAM_DIR)\DeploymentTargets\AzureFunctions\AzFunctionFiles\bin</DeploymentTargetBinPath>
+		<DeploymentTargetAzfunctPath>$(GX_PROGRAM_DIR)\DeploymentTargets\AzureFunctions\AzFunctionFiles\bin\.azurefunctions</DeploymentTargetAzfunctPath>
```
Comment (ggallotti) [inline src/AzureServerless/deploy.msbuild:12]: Naming Convenrions: Minor detail for next time.  'funct'
Please check this document for good naming conventions:
https://github.com/kettanaito/naming-cheatsheet

Specially: 
`Do not use contractions. They contribute to nothing but decreased readability of the code. Finding a short, descriptive name may be hard, but contraction is not an excuse for not doing so.`

File: src/AzureServerless/deploy.msbuild [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+			<ConfResources Include="$(DeployFullPath)\GeneXus.services" Condition ="Exists('$(DeployFullPath)\GeneXus.services')" />
+			<ConfResources Include="$(DeployFullPath)\functions.metadata"  />
+			<ConfResources Include="$(DeployFullPath)\appsettings.json" />
+		</ItemGroup>
+
+		<ItemGroup>
+			<Modules Include="@(Object -> '%(Module)')" />
+			<ModuleDirs Include="%(Modules.Identity)" />
+			<ModuleRelativePath Include="@(ModuleDirs->Replace('.', '\'))" />
+			<ModuleRelPathLower Include="@(ModuleRelativePath->ToLower())" />
```
Comment (ggallotti) [inline src/AzureServerless/deploy.msbuild:49]: Same here. ModuleRelativePathLower for more readability. 
https://github.com/kettanaito/naming-cheatsheet

File: src/AzureServerless/deploy.msbuild [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			
+	</Target>
+	
+	<Target Name="PrepareDeploy" DependsOnTargets="GenerateConfigs" >
+	<Message Text="Adding required files to the deploy...(this may take a while)" Importance="high"/>
+
+		<ItemGroup>
+			<ConfResources Include="$(DeployFullPath)\*.svc" />
+			<ConfResources Include="$(DeployFullPath)\Images.txt" Condition="Exists('$(DeployFullPath)\Images.txt')"/>
+			<ConfResources Include="$(DeployFullPath)\log.config" Condition="Exists('$(DeployFullPath)\log.config')"/>
```
Comment (ggallotti) [inline src/AzureServerless/deploy.msbuild:38]: Conf?  ConfigResources ? ;)

**Review by ggallotti — COMMENTED** [review]
LGTM

The only Issue is about: "AZURE_SERVERLESS_OPENAPI_FILE". 
Why this property is needed?

File: src/AzureServerless/Properties.xml (hunk 1 above)
Comment (sjuarezgx) [reply src/AzureServerless/Properties.xml:72]: There are two deployment targets
1. For Azure functions of different types (queue, http..) In this case the yaml is generated by the deploy engine, as you said.
2. This target (the one of the PR), is for the services of an app. In this case, I think it's better to ask the user to activate the Generate Open API property for all the objects of the deployment unit, specify, generate, and get de yaml file to pass it to the deploy. 
I wrote this in the spec: https://iwiki.genexus.com/wiki.aspx?29416,Spec+Azure+functions+de+tipo+Http#Dependencias
We can check it if you want.

File: src/AzureServerless/deploy.msbuild (hunk 2 above)
Comment (sjuarezgx) [reply src/AzureServerless/deploy.msbuild]: Perfect, thanks!

File: src/AzureServerless/deploy.msbuild (hunk 3 above)
Comment (sjuarezgx) [reply src/AzureServerless/deploy.msbuild:49]: OK

File: src/AzureServerless/deploy.msbuild (hunk 4 above)
Comment (sjuarezgx) [reply src/AzureServerless/deploy.msbuild:38]: OK, I'll take it into account. Thanks.

**Review by ggallotti — APPROVED** [review]

**Pull request merged**